### PR TITLE
Sils/prerename

### PR DESCRIPTION
**Pull request by Udayan12167**

### CODE REVIEW

File: coalib/results/result_actions/ShowPatchAction.py [hunk 1]
```diff
@@ -80,18 +80,22 @@ def apply(self,
         printer = ConsolePrinter(colored)
 
         for filename, this_diff in sorted(result.diffs.items()):
+            to_filename = (this_diff.rename if this_diff.rename else filename)
```
Comment (tushar-rishav) [inline coalib/results/result_actions/ShowPatchAction.py]: Why parenthesis?

**Comment by tushar-rishav** [comment]
LGTM

File: coalib/results/result_actions/OpenEditorAction.py [hunk 2]
```diff
@@ -20,7 +21,16 @@ class OpenEditorAction(ResultAction):
...
+        """
+        For being applicable, the result has to point to a number of files
+        that have to exist i.e. have not been previously deleted.
+        """
+        if not isinstance(result, Result) or not len(result.affected_code) > 0:
+            return False
+
+        filenames = set(src.renamed_file(file_diff_dict)
+                        for src in result.affected_code)
+        return all(exists(filename) for filename in filenames)
```
Comment (sils1297) [inline coalib/results/result_actions/OpenEditorAction.py]: sure we want all and not any?

File: coalib/results/result_actions/ShowPatchAction.py (hunk 1 above)
Comment (sils1297) [inline coalib/results/result_actions/ShowPatchAction.py]: please remove them, also in 84

File: coalib/results/result_actions/OpenEditorAction.py (hunk 2 above)
Comment (Udayan12167) [inline coalib/results/result_actions/OpenEditorAction.py]: Even if one of them doesn't exist we will get an error while opening files.

Comment (sils1297) [inline coalib/results/result_actions/OpenEditorAction.py]: we can just open only the files that exist and show a warning or so

Comment (sils1297) [inline coalib/results/result_actions/OpenEditorAction.py]: I agree that it's safer this way. Lets keep it.

**Comment by sils1297** [comment]
https://github.com/coala-analyzer/coala/pull/2186/commits/301ab20b5098c011559f64cb7a9b313d69a9fe21 needs work

**Comment by sils1297** [comment]
ack https://github.com/coala-analyzer/coala/pull/2186/commits/42323f066d09b838cfc263b1f898d07aa8ca63a2

**Comment by Udayan12167** [comment]
reack 0fe2fb7

File: coalib/results/result_actions/ShowPatchAction.py [hunk 3]
```diff
@@ -80,18 +80,22 @@ def apply(self,
...
         for filename, this_diff in sorted(result.diffs.items()):
+            to_filename = this_diff.rename if this_diff.rename else filename
+            to_filename = "/dev/null" if this_diff.delete else to_filename
             original_file = original_file_dict[filename]
             try:
                 current_file = file_diff_dict[filename].modified
-                new_file = (file_diff_dict[filename] + this_diff).modified
+                new_file = ([] if this_diff.delete
+                            else
+                            (file_diff_dict[filename] + this_diff).modified)
```
Comment (sils1297) [inline coalib/results/result_actions/ShowPatchAction.py]: actually modified should return `[]` on itself if the file has the deletion flag set so the if else is not needed

**Comment by sils1297** [comment]
ack https://github.com/coala-analyzer/coala/pull/2186/commits/57c9e09962ab40b33bda6b33812f9394370558d4

**Comment by sils1297** [comment]
@rultor merge

**Pull request merged**